components/InputField.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { InputHTMLAttributes, Ref, useEffect } from 'react';
+import { DeepMap, FieldError } from 'react-hook-form/dist/types';
+
+import styles from '../styles/InputField.module.scss';
+
+type InputFieldProps = {
+  label: string;
+  required?: boolean;
+  setValue: any;
+  register: any;
+} & InputHTMLAttributes<HTMLInputElement>;
+
+export default function InputField({
```

Comment:
Same as arrow

As for this whole function we can avoid heaps of re-renders by using the proper setup detailed in the first example
https://react-hook-form.com/get-started

since this is a UI lib component I would highly suggest using option 1 over option 3
https://react-hook-form.com/get-started#IntegratingwithUIlibraries

details on how to do input refs is here: (super useful information to have)
https://reactjs.org/docs/forwarding-refs.html